### PR TITLE
CORDA-939 Remove sslConfiguration from public constructor of CordaRPCClient

### PR DESCRIPTION
**Pull request by anthonykr3**
Remove the sslConfiguration paraemter from the public constructor of CordaRPCClient

Have created an internal factory method for initialising the sslConfiguration - and created an internal utility in net.corda.clienty.rpc.internal.CordaRPCClientUtils for other corda modules to get instances of CordaRPCClient. I considered making the constructor private but it seemed like it was used in so many places this broke the least things.

### CODE REVIEW

File: client/rpc/src/main/kotlin/net/corda/client/rpc/CordaRPCClient.kt [hunk 1]
```diff
@@ -72,9 +72,20 @@ data class CordaRPCClientConfiguration(val connectionMaxRetryInterval: Duration)
  */
 class CordaRPCClient @JvmOverloads constructor(
         hostAndPort: NetworkHostAndPort,
-        configuration: CordaRPCClientConfiguration = CordaRPCClientConfiguration.DEFAULT,
-        sslConfiguration: SSLConfiguration? = null
+        configuration: CordaRPCClientConfiguration = CordaRPCClientConfiguration.DEFAULT
 ) {
+    companion object {
+        internal fun getClientWithSsl(
```
Comment (shamsasari) [inline client/rpc/src/main/kotlin/net/corda/client/rpc/CordaRPCClient.kt]: `createWithSsl`?

File: client/rpc/src/main/kotlin/net/corda/client/rpc/CordaRPCClient.kt [hunk 2]
```diff
@@ -72,9 +72,20 @@ data class CordaRPCClientConfiguration(val connectionMaxRetryInterval: Duration)
...
+                hostAndPort: NetworkHostAndPort,
+                configuration: CordaRPCClientConfiguration = CordaRPCClientConfiguration.DEFAULT,
+                sslConfiguration: SSLConfiguration? = null
+        ) : CordaRPCClient{
+            return CordaRPCClient(hostAndPort, configuration).apply {
+                this.sslConfiguration = sslConfiguration
```
Comment (shamsasari) [inline client/rpc/src/main/kotlin/net/corda/client/rpc/CordaRPCClient.kt]: I think we can avoid this by having the default c'tor be private and which takes in this as a parameter and then the current default c'tor becomes a secondary c'tor.

File: client/rpc/src/main/kotlin/net/corda/client/rpc/internal/CordaRPCClientUtils.kt [hunk 3]
```diff
@@ -0,0 +1,13 @@
+package net.corda.client.rpc.internal
+
+import net.corda.client.rpc.CordaRPCClient
+import net.corda.client.rpc.CordaRPCClientConfiguration
+import net.corda.core.utilities.NetworkHostAndPort
+import net.corda.nodeapi.internal.config.SSLConfiguration
+
+/** Utility which exposes the internal Corda RPC constructor to other internal Corda components */
+fun getCordaRPCClientWithSsl(
```
Comment (shamsasari) [inline client/rpc/src/main/kotlin/net/corda/client/rpc/internal/CordaRPCClientUtils.kt]: The double-internal trick!

File: .ci/api-current.txt [hunk 4]
```diff
@@ -4166,10 +4180,12 @@ public final class net.corda.testing.node.User extends java.lang.Object
 public final class net.corda.client.rpc.CordaRPCClient extends java.lang.Object
   public <init>(net.corda.core.utilities.NetworkHostAndPort)
   public <init>(net.corda.core.utilities.NetworkHostAndPort, net.corda.client.rpc.CordaRPCClientConfiguration)
-  public <init>(net.corda.core.utilities.NetworkHostAndPort, net.corda.client.rpc.CordaRPCClientConfiguration, net.corda.nodeapi.internal.config.SSLConfiguration)
```
Comment (anthonykr3) [inline .ci/api-current.txt]: This is actually a backards compatibility change

File: client/rpc/src/main/kotlin/net/corda/client/rpc/CordaRPCClient.kt [hunk 5]
```diff
@@ -70,11 +70,24 @@ data class CordaRPCClientConfiguration(val connectionMaxRetryInterval: Duration)
  * @param configuration An optional configuration used to tweak client behaviour.
  * @param sslConfiguration An optional [SSLConfiguration] used to enable secure communication with the server.
  */
-class CordaRPCClient @JvmOverloads constructor(
+class CordaRPCClient private constructor(
         hostAndPort: NetworkHostAndPort,
         configuration: CordaRPCClientConfiguration = CordaRPCClientConfiguration.DEFAULT,
-        sslConfiguration: SSLConfiguration? = null
+        private var sslConfiguration: SSLConfiguration? = null
```
Comment (shamsasari) [inline client/rpc/src/main/kotlin/net/corda/client/rpc/CordaRPCClient.kt]: This doesn't need to be a property, i.e. no need for `private var`

File: .ci/api-current.txt (hunk 4 above)
Comment (shamsasari) [reply .ci/api-current.txt]: You mean backwards incompatible? Thankfully it's not when compared against V2: https://github.com/corda/corda/blob/release-V2/.ci/api-current.txt#L3233

Comment (anthonykr3) [reply .ci/api-current.txt]: Cracking

**Pull request merged**